### PR TITLE
Motor poten map integration

### PR DESCRIPTION
**Pull request by avnoks**

### CODE REVIEW

File: hal/motor_hal.hpp [hunk 1]
```diff
@@ -16,10 +16,14 @@ enum MotorStatus {
 class Motor {
 private:
     pthread_mutex_t lock_motor;
+    Coord r;
```
Comment (avnoks) [inline hal/motor_hal.hpp:19]: Would it be possible to add comments specifying what `r` and `theta` are?  If a description using words is too unwieldy, would it be possible to link to a figure?

Comment (Mixih) [reply hal/motor_hal.hpp:19]: +1, use doxygen compliant comments on fields.

File: robot/motors.hpp [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+#include <cstdint>
+
+// All measurements are in centimeters
+#define ROBOT_WIDTH 45.72
+#define WHEEL_DIAMETER 7.62
+
+typedef int32_t Speed;
+typedef int32_t AngVel;
+
+#endif
```
Comment (Mixih) [inline robot/motors.hpp:13]: add newline at EOF.

File: robot/poten_map.hpp [hunk 3]
```diff
@@ -52,3 +55,5 @@ Vector attr_poten_gradient(Vector cur_position, Vector goal);
  *   Repulsive potential gradient vector
  */
 Vector rep_poten_gradient(uint16_t * lidar);
+
+#endif
```
Comment (Mixih) [inline robot/poten_map.hpp:59]: add newline at EOF.

File: robot/motors.cpp [hunk 4]
```diff
@@ -1,3 +1,15 @@
...
 #include "hal.hpp"
+#include "poten_map.hpp"
+#include "motors.hpp"
 
-// implement me!
+Speed ang_vel_to_motor(AngVel ang_vel, Motor mtr) {
+  return WHEEL_DIAMETER * ang_vel * cos(mtr.theta() / mtr.r();
+}
+
+void velocity_to_motor_output(Speed speed, AngVel ang_vel) {
```
Comment (Mixih) [inline robot/motors.cpp:10]: needs to be plumbed.

**Review by Mixih — COMMENTED** [review]
Needs to be plumbed to event loop.

**Review by Mixih — APPROVED** [review]

**Pull request merged**